lib/std/mem.zig

```diff
@@ -357,18 +357,24 @@ pub fn eql(comptime T: type, a: []const T, b: []const T) bool {
     return true;
 }
 
-pub fn len(comptime T: type, ptr: [*:0]const T) usize {
+pub fn len(comptime T: type, ptr: var) usize {
+    const ptrInfo = @typeInfo(@TypeOf(ptr)).Pointer;
+    assert(ptrInfo.size == .Many);
+    const sentinel: T = comptime ptrInfo.sentinel.?;
+
     var count: usize = 0;
-    while (ptr[count] != 0) : (count += 1) {}
+    while (ptr[count] != sentinel) : (count += 1) {}
     return count;
 }
 
-pub fn toSliceConst(comptime T: type, ptr: [*:0]const T) [:0]const T {
-    return ptr[0..len(T, ptr) :0];
+pub fn toSliceConst(comptime T: type, ptr: var) [:@typeInfo(@TypeOf(ptr)).Pointer.sentinel.?]const T {
```

Comment:
It might make sense to remove `toSliceConst` entirely and have the return value have the `const`-ness of the argument?
Let me know if you want that as a followup PR.

Reply:
I agree with this. I would even go so far as to say it's worth having the parameter accept a `var` type in general to accommodate this.

Reply:
Created https://github.com/ziglang/zig/issues/3985 to track.